src/System.Device.Gpio/System/Device/Gpio/RaspberryBoardIdentification.cs

```diff
@@ -0,0 +1,337 @@
+            /// </summary>
+            Unknown,
+
+            /// <summary>
+            /// Processor is a BCM2708.
+            /// </summary>
+            Bcm2708,
+
+            /// <summary>
+            /// Processor is a BCM2709.
+            /// </summary>
+            Bcm2709,
+
+            /// <summary>
+            /// Processor is a BCM2710.
+            /// </summary>
+            Bcm2710,
+
+            /// <summary>
+            /// Processor is a11.
+            /// </summary>
+            Bcm2711,
+        }
+
+        #region Fields
+
+        private readonly Dictionary<string, string> _settings;
+
+        private RaspberryBoardIdentification(Dictionary<string, string> settings)
+        {
+            _settings = settings;
+        }
+
+        #endregion
+
+        #region Properties
+
+        /// <summary>
+        /// Get board model from firmware revision
+        /// See http://www.raspberrypi-spy.co.uk/2012/09/checking-your-raspberry-pi-board-version/ for information.
+        /// </summary>
+        /// <returns></returns>
+        internal Model GetBoardModel()
+        {
+            var firmware = Firmware;
+            switch (firmware & 0xFFFF)
+            {
+                case 0x2:
+                case 0x3:
+                    return Model.RaspberryPiBRev1;
+
+                case 0x4:
+                case 0x5:
+                case 0x6:
+                case 0xd:
+                case 0xe:
+                case 0xf:
+                    return Model.RaspberryPiBRev2;
+
+                case 0x7:
+                case 0x8:
+                case 0x9:
+                    return Model.RaspberryPiA;
+
+                case 0x10:
+                    return Model.RaspberryPiBPlus;
+
+                case 0x11:
+                    return Model.RaspberryPiComputeModule;
+
+                case 0x12:
+                    return Model.RaspberryPiAPlus;
+
+                case 0x1040:
+                case 0x1041:
+                    return Model.RaspberryPi2B;
+
+                case 0x0092:
+                case 0x0093:
+                    return Model.RaspberryPiZero;
+
+                case 0x00C1:
+                    return Model.RaspberryPiZeroW;
+
+                case 0x2082:
+                    return Model.RaspberryPi3B;
+
+                case 0x20D3:
+                    return Model.RaspberryPi3BPlus;
+
+                case 0x20A0:
+                    return Model.RaspberryPiComputeModule3;
+
+                case 0x3111:
+                    return Model.RaspberryPi4;
+
+                default:
+                    return Model.Unknown;
+            }
+        }
+
+        /// <summary>
+        /// Gets the processor name.
+        /// </summary>
+        /// <value>
+        /// The name of the processor.
+        /// </value>
+        public string ProcessorName
+        {
+            get
+            {
+                _settings.TryGetValue("Hardware", out string hardware);
```

Comment:
instead of strings could you just make those `public string Foo { get; private set; }` and fill the properties on construction directly rather than going through string?

Reply:
Sure (it's not so many we need for now, anyway). But do note that these properties are initially strings (we're parsing the /proc/cpuinfo file, which is basically a string=string property list).

Reply:
I'd still go with parsing this, we need only specific information and not generic knowledge database

Reply:
Now parsing in ctor and setting properties.